torchaudio/functional/functional.py

```diff
@@ -1861,6 +1863,21 @@ def mvdr_weights_souden(
     Returns:
         torch.Tensor: The complex-valued MVDR beamforming weight matrix with dimensions `(..., freq, channel)`.
     """
+    assert (
```

Comment:
since these assert statements are the same as for `rtf_power`, could you extract them into a common function to reuse?